client/src/pages/task/TaskPage.tsx

```diff
@@ -291,6 +297,7 @@ const TaskPage: React.FC = () => {
             {teacherControls} {buttons}
           </>
         }
+        onBack={goToAssignment}
```

Comment:
The button is also visible in the Task Pool now but does nothing. This should either be hidden or, even better, lead to the Task Pool list page.